for/data-frame construction forms

ralsei opened the pull request:
Closes #6. This PR does what was described there.

I also made the Scribble docs import `(for-label data-frame)`, which seems like it would be wrong, but it isn't, and adds hyperlinks to the Scribble docs for forms like `data-frame?`.

ralsei committed:
for/data-frame construction forms

alex-hhh commented:
Hi @ralsei , the build fails because there is a problem with the mirror site and downloading the racket installer fails.  Can you change the URL to "https://download.racket-lang.org/installers" to check if the build succeeds?  Thanks.

https://github.com/alex-hhh/data-frame/blob/6dd43cc900c90994f858dafa264f4e3f108e24a6/ci/install-racket.sh#L18

ralsei committed:
use generate-temporaries for column binders

this prevents having access to the vector inside the body, which is
really bad, since it's mostly filled with zeroes during iteration and
will not do what you want it to

ralsei committed:
change installer URL in install-racket
